src/loop.zig

```diff
@@ -98,21 +104,30 @@ pub const SingleThreaded = struct {
         }
     }
 
+    const Event = enum { js, zig };
+
+    fn eventsPtr(self: *Self, event: Event) *usize {
```

Comment:
Pretty sure if you mark `event` as comptime (here and all the other places), it'll eliminate the runtime switch.